src/bus/index.spec.js

```diff
@@ -0,0 +1,265 @@
+          );
+        });
+      });
+    });
+
+    context(
+      'when a websocket does not already exist for the organization',
+      function() {
+        context('when successfully connecting to the message bus', function() {
+          let bus;
+          let expectedApiToken;
+          let promise;
+          let sdk;
+
+          beforeEach(function() {
+            expectedApiToken = faker.internet.password();
+
+            sdk = {
+              ...baseSdk,
+              auth: {
+                ...baseSdk.auth,
+                getCurrentApiToken: this.sandbox
+                  .stub()
+                  .resolves(expectedApiToken)
+              }
+            };
+
+            bus = new Bus(sdk, baseRequest);
+            bus._baseSocketUrl = expectedHost;
+
+            promise = bus.connect(expectedOrganization.id);
+          });
+
+          context('when intially opening the websocket', function() {
+            it('gets an api token', function() {
+              return promise.then(() => {
+                expect(sdk.auth.getCurrentApiToken).to.be.calledWith(
+                  'contxtAuth'
+                );
+              });
+            });
+
+            it('stores a copy of the websocket', function() {
+              return promise.then(() => {
+                const socket = bus._sockets[expectedOrganization.id];
+
+                expect(socket).to.be.an.instanceof(Socket);
+                expect(socket._organizationId).to.equal(
+                  expectedOrganization.id
+                );
+                expect(socket._socket.url).to.equal(
+                  `${expectedHost}/organizations/${
+                    expectedOrganization.id
+                  }/stream`
+                );
+              });
+            });
+          });
+
+          context('when the websocket closes', function() {
+            beforeEach(function() {
+              return promise.then(() => {
+                server.close();
+              });
+            });
+
+            it('clears out the stored copy of the websocket when it closes', function() {
+              return promise.then(() => {
+                expect(bus._sockets[expectedOrganization.id]).to.be.null;
+              });
+            });
+          });
+        });
+
+        context(
+          'when unsuccessful at connecting to the message bus',
+          function() {
+            let bus;
+            let expectedApiToken;
+            let promise;
+            let sdk;
+            let socketUrl;
+
+            beforeEach(function() {
+              expectedApiToken = faker.internet.password();
+
+              sdk = {
+                ...baseSdk,
+                auth: {
+                  ...baseSdk.auth,
+                  getCurrentApiToken: this.sandbox
+                    .stub()
+                    .resolves(expectedApiToken)
+                }
+              };
+
+              socketUrl = `wss://${faker.internet.domainName()}`;
```

Comment:
does this connect to a real URL during the test?

Reply:
The mock websocket server is created using `${expectedHost}/organizations/${expectedOrganization.id}/stream` and for the successful connections, it's using that URL to connect. This test is testing for an unsuccessful connection so it's setting the client connection URL to a value that's not the server so that way there shouldn't be a successful connection

Reply:
I gotcha, my worry is that it's still throwing a request out into the webosphere. Any way to make it not even try to connect to a real server?

Reply:
It's not making a request to the real world -- the `ws` module is replaced by the `mock-socket` module at the top of this file